### PR TITLE
docs/protocol/specifications: specify Merkle Patricia Tree as a set

### PR DESCRIPTION
**Pull request by kr**

### CODE REVIEW

**Comment by kr** [comment]
PTAL @oleganza @jbowens

**Comment by kr** [comment]
I think I also need to update https://github.com/chain/chain/blob/main/docs/protocol/specifications/data.md#assets-merkle-root in this same PR.

**Comment by kr** [comment]
PTAL

File: docs/protocol/specifications/data.md [hunk 1]
```diff
@@ -506,29 +502,25 @@ Note that we do not require the length of the input list to be a power of two. T
...
-
+    MPTH({d(0)}) = SHA3-256(0x00 || d(0))
 
+For n > 1, let the bit string p be the longest common prefix of all items in D[n], and let k be the number of items that have a prefix `p||0` (that is, p concatenated with the single bit 0). The merkle patricia tree hash of an n-element list `D[n]` is then defined recursively as:
 
+    MPTH(D[n]) = SHA3-256(0x01 || MPTH(D[0:k]) || MPTH(D[k:n]))
 
+where `||` is concatenation and `D[k1:k2]` denotes the list `{d(k1), d(k1+1),..., d(k2-1)}` of length `(k2 - k1)`. (Note that the hash calculations for leaves and nodes differ. This domain separation is required to give second preimage resistance.)
 
+Note that the resulting merkle patricia tree may not be balanced; however, its shape is uniquely determined by the number of leaves.
```
Comment (oleganza) [inline docs/protocol/specifications/data.md]: the shape is uniquely determined by _both_ the number of leaves and their content.

**Comment by kr** [comment]
Addressed feedback, PTAL.

File: docs/protocol/specifications/data.md [hunk 2]
```diff
@@ -464,11 +464,7 @@ Root hash of the [merkle binary hash tree](#merkle-binary-tree) formed by the [t
 
 Root hash of the [merkle patricia tree](#merkle-patricia-tree) formed by unspent outputs with an **asset version 1** after applying the block. Allows bootstrapping nodes from recent blocks and an archived copy of the corresponding merkle patricia tree without processing all historical transactions.
 
-The tree contains [non-retired](#retired-asset) unspent outputs (one or more per [asset ID](#asset-id)) where both key and value are the same value — the [Output ID](#output-id) of the unspent output.
-
-Key                     | Value
-------------------------|------------------------------
-[Output ID](#output-id) | [Output ID](#output-id)
+The tree contains the list of the [Output IDs](#output-id) of all [non-retired](#retired-asset) unspent outputs (one or more per [asset ID](#asset-id)).
```
Comment (oleganza) [inline docs/protocol/specifications/data.md]: I think my note on "one or more per asset ID" was redundant and distracting.

**Pull request merged**